File: src/main/java/com/aquasecurity/plugins/trivy/actions/ShowTrivySettingsAction.java
package com.aquasecurity.plugins.trivy.actions;

import com.aquasecurity.plugins.trivy.settings.TrivySettingsConfigurable;
import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.options.ShowSettingsUtil;
import com.intellij.openapi.project.Project;
import org.jetbrains.annotations.NotNull;

public class ShowTrivySettingsAction extends AnAction {


    @Override
    public void update(@NotNull AnActionEvent e) {
        super.update(e);
    }

    @Override
    public void actionPerformed(@NotNull AnActionEvent e) {

        Project project = e.getProject();
        if (project == null) {
            return;
        }

        TrivySettingsConfigurable configurable = new TrivySettingsConfigurable();
        ShowSettingsUtil.getInstance().editConfigurable(project, configurable);
    }
}

File: src/main/kotlin/com/aquasecurity/plugins/trivy/actions/ShowTrivySettingsAction.kt
package com.aquasecurity.plugins.trivy.actions

import com.intellij.openapi.actionSystem.ActionUpdateThread
import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.options.ShowSettingsUtil

class ShowTrivySettingsAction : AnAction() {
  override fun update(e: AnActionEvent) {
    super.update(e)
  }

  override fun actionPerformed(e: AnActionEvent) {
    val project = e.project ?: return

    ShowSettingsUtil.getInstance().showSettingsDialog(project, "Trivy: Settings")
  }

  override fun getActionUpdateThread(): ActionUpdateThread {
    return super.getActionUpdateThread()
  }
}
